Mathlib/FieldTheory/PerfectClosure.lean

```diff
@@ -393,17 +393,7 @@ theorem frobenius_mk (x : ℕ × K) :
     (frobenius (PerfectClosure K p) p : PerfectClosure K p → PerfectClosure K p) (mk K p x) =
       mk _ _ (x.1, x.2 ^ p) := by
   simp only [frobenius_def]
-  cases' x with n x
-  dsimp only
-  suffices ∀ p' : ℕ, mk K p (n, x) ^ p' = mk K p (n, x ^ p') by apply this
-  intro p
-  induction p with
-  | zero => apply R.sound; rw [(frobenius _ _).iterate_map_one, pow_zero]
-  | succ p ih =>
-    rw [pow_succ, ih]
-    symm
-    apply R.sound
-    simp only [pow_succ, iterate_map_mul]
+  exact mk_pow K p x p
```

Comment:
Maybe `simp only [frobenius_def]` is also unnecessary; you can write all of them into one line: `... := mk_pow K p x p` is enough. (untested)

Reply:
Yes, that works too, but the existence of the `frobenius_def` lemma suggests to me that maybe it's better to avoid depending on definitional equality for `frobenius`.

Reply:
Yes, let's keep it.